docs/element-definitions.md

```diff
@@ -1353,6 +1356,339 @@ the `inferred` link tells the user that the element was created based on some
+
+### Basic Authentication Scheme (Base API Element)
+
+This element may be used to define a basic authentication scheme implementation for an API as described in [RFC 2617](https://tools.ietf.org/html/rfc2617).
+
+The element MAY have a username and password defined as member elements within the scheme, but these are not required.
+
+- `username` (string, optional)
+- `password` (string, optional)
+
+#### Properties
+
+- `element`: Basic Authentication Scheme (string, fixed)
+- `content` (array[Member Element])
+
+#### Example
+
+This example shows a custom basic authentication scheme being defined as `Custom Basic Auth`. This scheme is then used on an HTTP transaction within a resource. Please note this example is incomplete for the sake of keeping it short.
+
+```json
+{
+  "element": "category",
+  "meta": {
+    "classes": ["api"]
+  },
+  "content": [
+    {
+      "element": "category",
+      "meta": {
+        "classes": ["authSchemes"]
+      },
+      "content": [
+        {
+          "element": "Basic Authentication Scheme",
+          "meta": {
+            "id": "Custom Basic Auth"
+          },
+          "content": [
+            {
+              "element": "member",
+              "content": {
+                "key": {
+                  "element": "string",
+                  "content": "username"
+                },
+                "value": {
+                  "element": "string",
+                  "content": "john.doe"
+                }
+              }
+            },
+            {
+              "element": "member",
+              "content": {
+                "key": {
+                  "element": "string",
+                  "content": "password"
+                },
+                "value": {
+                  "element": "string",
+                  "content": "1234password"
+                }
+              }
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "element": "resource",
+      "attributes": {
+        "href": "/users"
+      },
+      "content": [
+        {
+          "element": "transition",
+          "content": [
+            {
+              "element": "httpTransaction",
+              "attributes": {
+                "authSchemes": [
+                  {
+                    "element": "Custom Basic Auth"
+                  }
+                ]
+              }
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+### Token Authentication Scheme (Base API Element)
+
+This describes an authentication scheme that uses a token as a way to authenticate and authorize. The token MAY exist as an HTTP header field or a query parameter.
+
+- One of
+    - `httpHeaderName` (string)
+    - `queryParameterName` (string)
+
+When used as a query parameter, an HREF Variable is not required to be defined within the scope of the resource or transition, but is rather infered from the used token authentications scheme.
```

Comment:
I think we should keep this line here for a lot of reasons:

1. We don't want to change anything the user has written in the api description document and should represent it correctly in refract
2. A lot of users want authentication in API Blueprint because they are tired of adding parameter/header everytime.
3. We should leave the interpretation of the authentication to the subsequent tooling. We could not do this if do the interpretation ourselves.